jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunction.kt

```diff
@@ -6,26 +6,80 @@ package software.aws.toolkits.jetbrains.services.lambda.upload
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.project.Project
 import icons.AwsIcons
 import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationTemplateIndex
 import software.aws.toolkits.jetbrains.services.iam.IamRole
 import software.aws.toolkits.jetbrains.services.lambda.runtime
 import software.aws.toolkits.jetbrains.services.lambda.upload.EditFunctionMode.NEW
 import software.aws.toolkits.resources.message
 
-class CreateLambdaFunction(private val handlerName: String? = null) : AnAction(message("lambda.create_new"), null, AwsIcons.Actions.LAMBDA_FUNCTION_NEW) {
+open class CreateLambdaFunction() : AnAction(message("lambda.create_new"), null, AwsIcons.Actions.LAMBDA_FUNCTION_NEW) {
     override fun actionPerformed(event: AnActionEvent) {
         val project = event.getRequiredData(LangDataKeys.PROJECT)
         val runtime = event.runtime()
 
-        val dialog = if (handlerName != null) {
-            EditFunctionDialog(project = project, mode = NEW, runtime = runtime, handlerName = handlerName)
-        } else {
-            EditFunctionDialog(project = project, mode = NEW, runtime = runtime)
-        }
+        val dialog = getEditFunctionDialog(project, runtime)
 
         dialog.show()
     }
+
+    protected open fun getEditFunctionDialog(project: Project, runtime: Runtime?): EditFunctionDialog =
+        EditFunctionDialog(project = project, mode = NEW, runtime = runtime)
+}
+
+open class CreateElementBasedLambdaFunction(private val handlerName: String, private val project: Project)
```

Comment:
I don't understand what these are trying to achieve. How does this actually suppress the action from being visible?

Reply:
The change doesn't affect visibility on `CreateLambdaFunction` , but makes it so that `plugin.xml` is the only thing directly referencing that class (which doesn't have visibility requirements at this time). The derived class has visibility suppression, and the gutter icon code works with the derived class(es)